rbd/watchers_mimic_test.go

```diff
@@ -0,0 +1,116 @@
+// +build !luminous
+//
+// Ceph Mimic is the first version that supports watchers through librbd.
+
+package rbd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestListWatchers(t *testing.T) {
+	conn := radosConnect(t)
+	require.NotNil(t, conn)
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	require.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+
+	name := GetUUID()
+	options := NewRbdImageOptions()
+	err = CreateImage(ioctx, name, 1<<22, options)
+	require.NoError(t, err)
+
+	t.Run("imageNotOpen", func(t *testing.T) {
+		image, err := OpenImage(ioctx, name, NoSnapshot)
+		require.NoError(t, err)
+		require.NotNil(t, image)
+
+		err = image.Close()
+		require.NoError(t, err)
+
+		_, err = image.ListWatchers()
+		assert.Equal(t, ErrImageNotOpen, err)
+	})
+
+	// open image read-only, as OpenImage() automatically adds a watcher
+	image, err := OpenImageReadOnly(ioctx, name, NoSnapshot)
+	require.NoError(t, err)
+	require.NotNil(t, image)
+
+	t.Run("noWatchers", func(t *testing.T) {
+		watchers, err := image.ListWatchers()
+		assert.NoError(t, err)
+		assert.Equal(t, 0, len(watchers))
+	})
+
+	var image2, image3 *Image
```

Comment:
ditto.

Reply:
Sorry, I don't understand how you mean this either.